### DISCUSSION TITLE
NPM How to update "resolved" with custom registry

### DISCUSSION THREAD

**Discussion by arossert** [renovatebot/renovate#25334] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.31.1

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I'm using renovatebot in order to automatically create PRs with new versions of modules in my nodejs application.
I'm also using a custom cache registry so the `"resolved"` part of the package-lock.json is pointing to this local registry, once renovate created the PR, it changes it to the default `https://registry.npmjs.org` one.

Is there a way to configure it so it will keep using my registry?

### What you tried so far.

I tried to use `registryUrls` and `npmrc` in the `renovate.json` file

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
Sounds like you need to set `registry=.....` either in a `.npmrc` file in the repo, or in `npmrc` within Renovate. If you've specified correctly, Renovate should not be looking up the default registry

**Comment by arossert** [comment]
@rarkins I created `.npmrc` file inside the repo and set the registry key
```
registry=https://artifactory.sbops.com/artifactory/api/npm/sbmain-npm-dev/
```

Even after doing this the PR that was created changes the `resolved` property to use the default registry

![image](https://github.com/renovatebot/renovate/assets/7085494/d8b55580-c930-496f-a060-320251c69596)

**Reply by rarkins** [reply]
Please run Renovate against just that repo and share the logs. Take a look for the string `npmrc` in the logs

It is _npm_ which updates those fields, not Renovate. If npm points them to the default registry it's because it's not seeing the `.npmrc` file in the place it's expecting.

**Comment by arossert** [comment]
I removed `"npmrcMerge": true` from the `renovate.json` file and it seems to fix this issue, Tx.
